test/e2e/providers/azure/azure.go

```diff
@@ -119,11 +124,218 @@ func (p *Provider) GetType() config.PlatformType {
 }
 
 func (p *Provider) StorageSupport() bool {
-	return false
+	return true
 }
 
-func (p *Provider) CreatePVC(_ client.Interface, _ string, _ *core.PersistentVolume) (*core.PersistentVolumeClaim, error) {
-	return nil, fmt.Errorf("storage not supported on azure")
+func (p *Provider) CreatePVC(c client.Interface, namespace string, _ *core.PersistentVolume) (*core.PersistentVolumeClaim, error) {
+	if err := p.ensureWindowsCSIDaemonSet(c); err != nil {
+		return nil, err
+	}
+	storageClassName := "azurefile-csi"
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "e2e" + "-",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteMany},
+			Resources: core.ResourceRequirements{
+				Requests: core.ResourceList{core.ResourceStorage: resource.MustParse("2Gi")},
```

Comment:
nit: do we want to request less storage for the e2e suite?

Reply:
I thought 2Gi was already pretty small. Also not sure how much customers will have.

Reply:
2Gi is small in today's world 😄